### ISSUE TITLE
Fail when no linters are passed

### ISSUE THREAD

**Issue by AlexSkilled** [golangci/golangci-lint#6321]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

go install

### Description of the problem

When config with empty linters list is passed - golangci-lint run fails with typecheck errors.

I was expecting lint to pass without errors bc there is no rules to break, but got quite the opposite result

The code I run linter on:

```
package main

import "fmt"

func main() {
    fmt.Println("Hello World!")
}

```

Tested on mac\linux\windows\CICD gitlab pipeline

Everywhere is the same problem

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
# golangci-lint has version 2.8.0 built with go1.25.5 from e2e40021 on 2026-01-07T21:29:47Z
```

</details>

### Configuration

<details>

```console
version: '2'

run:
  timeout: 5m
  concurrency: 4
  issues-exit-code: 42

formatters:
  exclusions:
    paths: []


linters:
  default: none
  enable: []

  settings:
    varnamelen:
      max-distance: 25
      min-name-length: 2
      ignore-type-assert-ok: true
      check-type-param: true
      ignore-chan-recv-ok: true
    govet:
      enable:
        - fieldalignment
  exclusions:
    paths:
      - configs
    rules:
      - linters:
          - staticcheck
        text: "SA9003:"
      - linters:
          - lll
        source: "^//go:generate "
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.25.5 darwin/arm64
AR='ar'
CC='cc'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='1'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='c++'
GCCGO='gccgo'
GO111MODULE=''
GOARCH='arm64'
GOARM64='v8.0'
GOAUTH='netrc'
GOBIN='/Users/alexbukov/go/bin'
GOCACHE='/Users/alexbukov/Library/Caches/go-build'
GOCACHEPROG=''
GODEBUG=''
GOENV='/Users/alexbukov/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS=''
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/jz/z6z0svbn0jvbjsw57v3l8_600000gn/T/go-build3662516021=/tmp/go-build -gno-record-gcc-switches -fno-common'
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMOD='/Users/alexbukov/tst/go.mod'
GOMODCACHE='/Users/alexbukov/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='darwin'
GOPATH='/Users/alexbukov/go'
GOPRIVATE=''
GOPROXY='direct'
GOROOT='/opt/homebrew/Cellar/go/1.25.5/libexec'
GOSUMDB='off'
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/alexbukov/Library/Application Support/go/telemetry'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/opt/homebrew/Cellar/go/1.25.5/libexec/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.25.5'
GOWORK=''
PKG_CONFIG='pkg-config'

```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO golangci-lint has version 2.8.0 built with go1.25.5 from e2e40021 on 2026-01-07T21:29:47Z 
INFO [config_reader] Config search paths: [./ /Users/alexbukov/tst /Users/alexbukov /Users /] 
INFO [config_reader] Used config file .golangci.yaml 
INFO [config_reader] Module name "tst"            
INFO [goenv] Read go env for 6.956792ms: map[string]string{"GOCACHE":"/Users/alexbukov/Library/Caches/go-build", "GOROOT":"/opt/homebrew/Cellar/go/1.25.5/libexec"} 
INFO [lintersdb] Active 0 linters: []             
INFO [loader] Go packages loading at mode 8199 (files|name|compiled_files) took 186.40725ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 564.5µs 
INFO [linters_context/goanalysis] analyzers took 206.459µs with top 10 stages: typecheck: 206.459µs 
INFO [runner/exclusion_paths] Skipped 0 issues by pattern "configs" 
INFO [runner/exclusion_rules] Skipped 0 issues by rules: [Text: "SA9003:", Linters: "staticcheck"] 
INFO [runner/exclusion_rules] Skipped 0 issues by rules: [Source: "^//go:generate ", Linters: "lll"] 
INFO [runner] Processors filtering stat (in/out): max_from_linter: 1/1, sort_results: 1/1, nolint_filter: 1/1, fixer: 1/1, source_code: 1/1, path_shortener: 1/1, severity-rules: 1/1, path_prettifier: 1/1, cgo: 1/1, filename_unadjuster: 1/1, exclusion_paths: 1/1, generated_file_filter: 1/1, path_relativity: 1/1, exclusion_rules: 1/1, max_per_file_from_linter: 1/1, max_same_issues: 1/1, path_absoluter: 1/1, invalid_issue: 1/1, diff: 1/1, uniq_by_line: 1/1 
INFO [runner] processing took 37.712µs with stages: source_code: 20.916µs, exclusion_rules: 4.625µs, exclusion_paths: 4.583µs, uniq_by_line: 2.458µs, sort_results: 1.542µs, path_relativity: 1.041µs, max_same_issues: 583ns, path_shortener: 417ns, max_from_linter: 334ns, nolint_filter: 209ns, fixer: 167ns, filename_unadjuster: 126ns, path_absoluter: 124ns, severity-rules: 84ns, path_prettifier: 84ns, cgo: 84ns, generated_file_filter: 84ns, invalid_issue: 84ns, max_per_file_from_linter: 84ns, diff: 83ns 
INFO [runner] linters took 2.071917ms with stages: typecheck: 1.980833ms 
main.go:3:8: could not import fmt (Config.Importer.Import(fmt) returned nil but no error) (typecheck)
import "fmt"
       ^
1 issues:
* typecheck: 1
INFO File cache stats: 1 entries of total size 76B 
INFO Memory: 3 samples, avg is 35.1MB, max is 35.1MB 
INFO Execution took 196.489917ms  
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
> I was expecting lint to pass without errors bc there is no rules to break, but got quite the opposite result

When there is no linter, there is no lint.
It doesn't make a lot of sense to do what you are doing.
I will improve the error, but it will still have an error.
